lib/savings_transformation.rb

```diff
@@ -39,4 +39,16 @@ def someone_over_61?(application)
   def applicant_age_or_zero(application)
     application.applicant.date_of_birth.present? ? application.applicant.age : 0
   end
+
+  def savings_investment_valid?(application)
+    result = false
+    if application.threshold_exceeded == false ||
+       (
+         application.threshold_exceeded &&
+         (application.partner_over_61 && application.high_threshold_exceeded == false)
+       )
+      result = true
+    end
+    result
```

Comment:
This whole method could turn out just to the 'if statement condition' with the same result:

```ruby
application.threshold_exceeded == false ||
  (
    application.threshold_exceeded &&
    (application.partner_over_61 && application.high_threshold_exceeded == false)
  )
```